portfolio/src/main/java/com/google/sps/servlets/ChartServlet.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/chart-data")
+public class ChartServlet extends HttpServlet {
+
+  // Stores data in the form "country_name: rice_produced."
+  private LinkedHashMap<String, Double> countryMap = new LinkedHashMap<>();
+
+  @Override
+  public void init() {
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/edible_food_2011.csv"));
+
+    String headerLine1 = scanner.nextLine();
+    String headerLine2 = scanner.nextLine();
+    String headerLine3 = scanner.nextLine();
+
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      String[] cells = line.split(",");
+
+      String country = String.valueOf(cells[1]);
```

Comment:
Suggest creating variables for these numbers for more readability, i.e.:
int countryColumn = 1;

Reply:
you should also verify the the length of the string array before accessing via hardcoded numbers